scala-package/examples/src/main/scala/ml/dmlc/mxnet/examples/neuralstyle/NeuralStyle.scala

```diff
@@ -0,0 +1,246 @@
+      val lr = new FactorScheduler(step = 10, factor = 0.9f)
+
+      saveImage(contentNp, s"${alle.outputDir}/input.jpg", alle.guassianRadius)
+      saveImage(styleNp, s"${alle.outputDir}/style.jpg", alle.guassianRadius)
+//      val optimizer = new SGD(
+//          learningRate = alle.lr,
+//          momentum = 0.9f,
+//          wd = 0.005f,
+//          clipGradient = 10,
+//          lrScheduler = lr)
+       val optimizer = new Adam(
+          learningRate = alle.lr,
+          wd = 0.005f,
+          clipGradient = 10,
+          lrScheduler = lr)
+      val optimState = optimizer.createState(0, img)
+
+      logger.info(s"start training arguments $alle")
+
+      var oldImg = img.copy()
+      var gradArray: Array[NDArray] = null
+      var eps = 0f
+      var trainingDone = false
+      var e = 0
+      while(e < alle.maxNumEpochs && !trainingDone) {
+        modelExecutor.data.set(img)
+        modelExecutor.executor.forward()
+
+        // style gradient
+        for(i <- 0 until modelExecutor.style.length) {
+          val gram = gramExecutor(i)
+          gram.data.set(modelExecutor.style(i))
+          gram.executor.forward()
+          gram.executor.backward(gram.executor.outputs(0) - styleArray(i))
+          val tmp = gram.data.shape(1) * gram.data.shape(1) *
+            gram.data.shape(2) * gram.data.shape(3)
+          gram.dataGrad.set(gram.dataGrad / tmp.toFloat)
+          gram.dataGrad.set(gram.dataGrad * alle.styleWeight)
+        }
+
+        // content gradient
+        contentGrad.set((modelExecutor.content - contentArray) * alle.contentWeight)
+
+        // image gradient
+        gradArray = gramExecutor.map(_.dataGrad) :+ contentGrad
+        modelExecutor.executor.backward(gradArray)
+
+        optimizer.update(0, img, modelExecutor.dataGrad, optimState)
+        eps = twoNorm((oldImg - img).toArray) / twoNorm(img.toArray)
+        oldImg.set(img)
+        logger.info(s"epoch $e, relative change $eps")
+
+        if(eps < alle.stopEps) {
+          logger.info("eps < args.stop_eps, training finished")
+          trainingDone = true
+        }
+        if((e + 1) % alle.saveEpochs == 0) {
+          saveImage(img, s"${alle.outputDir}/tmp_${e + 1}.jpg", alle.guassianRadius)
+        }
+        e = e + 1
+      }
+      saveImage(img, s"${alle.outputDir}/out.jpg", alle.guassianRadius)
+      logger.info("Finish fit ...")
+    } catch {
+      case ex: Exception => {
+        logger.error(ex.getMessage, ex)
+        parser.printUsage(System.err)
+        sys.exit(1)
+      }
+    }
+  }
+}
+
+class NeuralStyle {
+  @Option(name = "--model", usage = "the pretrained model to use: ['vgg']")
+  private val model: String = "vgg19"
+  @Option(name = "--content-image", usage = "the content image")
+  private val contentImage: String = "input/IMG_4343.jpg"
+  @Option(name = "--style-image", usage = "the style image")
+  private val styleImage: String = "input/starry_night.jpg"
```

Comment:
instead of providing default arguments 'input/IMG_4343.jpg' and 'starry_night.jpg', a better way is to always let users to specify them and assert(contentImage != null && styleImage != null)